### ISSUE TITLE
detection_output_layer.cu:123] Couldn't find any detections

### ISSUE THREAD

**Issue by sunwooOh** [weiliu89/caffe#230]
Hi,
I'm trying to fine-tune from the reduced VGG model, using single-object("text" or "non-text") detection dataset.
I modified ssd_pascal.py for that, and when I run `ssd_detect.ipynb`, I get this kind of error:

> I1017 17:57:03.871824  4738 net.cpp:228] relu4_3 does not need backward computation.
> I1017 17:57:03.871832  4738 net.cpp:228] conv4_3 does not need backward computation.
> I1017 17:57:03.871840  4738 net.cpp:228] relu4_2 does not need backward computation.
> I1017 17:57:03.871847  4738 net.cpp:228] conv4_2 does not need backward computation.
> I1017 17:57:03.871855  4738 net.cpp:228] relu4_1 does not need backward computation.
> I1017 17:57:03.871863  4738 net.cpp:228] conv4_1 does not need backward computation.
> I1017 17:57:03.871872  4738 net.cpp:228] pool3 does not need backward computation.
> I1017 17:57:03.871881  4738 net.cpp:228] relu3_3 does not need backward computation.
> I1017 17:57:03.871889  4738 net.cpp:228] conv3_3 does not need backward computation.
> I1017 17:57:03.871897  4738 net.cpp:228] relu3_2 does not need backward computation.
> I1017 17:57:03.871906  4738 net.cpp:228] conv3_2 does not need backward computation.
> I1017 17:57:03.871913  4738 net.cpp:228] relu3_1 does not need backward computation.
> I1017 17:57:03.871922  4738 net.cpp:228] conv3_1 does not need backward computation.
> I1017 17:57:03.871932  4738 net.cpp:228] pool2 does not need backward computation.
> I1017 17:57:03.871939  4738 net.cpp:228] relu2_2 does not need backward computation.
> I1017 17:57:03.871948  4738 net.cpp:228] conv2_2 does not need backward computation.
> I1017 17:57:03.871956  4738 net.cpp:228] relu2_1 does not need backward computation.
> I1017 17:57:03.871965  4738 net.cpp:228] conv2_1 does not need backward computation.
> I1017 17:57:03.871974  4738 net.cpp:228] pool1 does not need backward computation.
> I1017 17:57:03.871984  4738 net.cpp:228] relu1_2 does not need backward computation.
> I1017 17:57:03.871991  4738 net.cpp:228] conv1_2 does not need backward computation.
> I1017 17:57:03.872001  4738 net.cpp:228] relu1_1 does not need backward computation.
> I1017 17:57:03.872009  4738 net.cpp:228] conv1_1 does not need backward computation.
> I1017 17:57:03.872020  4738 net.cpp:228] data_input_0_split does not need backward computation.
> I1017 17:57:03.872028  4738 net.cpp:228] input does not need backward computation.
> I1017 17:57:03.872035  4738 net.cpp:270] This network produces output detection_out
> I1017 17:57:03.872122  4738 net.cpp:283] Network initialization done.
> I1017 17:57:04.416875  4738 net.cpp:761] Ignoring source layer data
> I1017 17:57:04.416931  4738 net.cpp:761] Ignoring source layer data_data_0_split
> I1017 17:57:04.432705  4738 net.cpp:761] Ignoring source layer mbox_loss
> I1017 17:57:04.912191  4738 detection_output_layer.cu:123] Couldn't find any detections

I have checked that I have correct ground truth labels and image data by running `ssd.ipynb`.
What do you guys think is the problem? Hope I get some help.

[deploy.txt](https://github.com/weiliu89/caffe/files/533021/deploy.txt)
(uploaded as txt since github doesn't support prototxt file format)

**Comment by weiliu89** [comment]
That is because the detector cannot find any detections which have score higher than the threshold (0.01).

**Comment by sunwooOh** [comment]
Hi, @weiliu89. Thanks for your reply.
In the case that all detections have score less than 0.01, is there any way to train this network?
Should I change that threshold 0.01 to much lower value?

**Comment by sunwooOh** [comment]
Nope. Unfortunately, still doesn't generate any detections even though I changed the threshold to 0.

**Comment by dongmingyang** [comment]
My dataset is make up with half gray and half RGB, so i changed the "force_color = True" ,then i got this problem.
The problem went away when i changed it back (force_color = False) and trained again with just RGB data.

Hope this helpful :）
